packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolveFactory_def.hpp

```diff
@@ -299,10 +272,20 @@ void BelosLinearOpWithSolveFactory<Scalar>::setParameterList(
   )
 {
   TEUCHOS_TEST_FOR_EXCEPT(paramList.get()==NULL);
-  paramList->validateParametersAndSetDefaults(*this->getValidParameters(), 1);
+  std::string solverName = paramList->get<std::string>(SolverType_name, SolverType_default);
+  std::string solverNameUC = Belos::Impl::upperCase (solverName);
```

Comment:
It looks like you need Belos' canonical solver name here.  Belos does more than just capitalize the input name; it also has a mapping from "alias(es)" to a "canonical name."  If you want access to that here, let me know.

Reply:
This should be ok, since I am actually getting valid lists for all available solver names, even if they are just aliases.